### DISCUSSION TITLE
How to disable trimming of blank lines?

### DISCUSSION THREAD

**Discussion by trbabb** [zed-industries/zed#18558] [Feature Requests]
Hi, apologies if this is in the documentation and I missed it—

I notice that by default (in C++, at least) the editor is configured to automatically delete whitespace in otherwise empty lines. Code style in my codebase is to preserve indentation in empty lines. Is there a way to disable this feature?

Thanks!

**Comment by trbabb** [comment]
Bump for this. Is this possible?

**Comment by bemky** [comment]
Bump for me too. I can't find another discussion about this. I'm bringing an existing project to Zed that is full of lines that are only whitespace (poor linting before). Now every saved file has tons of whitespace only changes to files.

**Comment by roworu** [comment]
```
  "languages": {
    "C++": {
      "format_on_save": "off",
      "remove_trailing_whitespace_on_save": false
    }
  }
```

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links
